share/rocm/cmake/ROCMInstallTargets.cmake

```diff
@@ -7,10 +7,67 @@ include(GNUInstallDirs)
 include(ROCMPackageConfigHelpers)
 
 set(ROCM_INSTALL_LIBDIR lib)
+if(WIN32)
+    set(ROCM_USE_DEV_COMPONENT OFF CACHE BOOL "Generate a devel package?")
+else()
+    set(ROCM_USE_DEV_COMPONENT ON CACHE BOOL "Generate a devel package?")
+endif()
+
+function(rocm_install)
+    if(ARGV0 STREQUAL "TARGETS")
+        # rocm_install_targets deals with the component in its own fashion.
+        rocm_install_targets("${ARGN}")
+    elseif(NOT ROCM_USE_DEV_COMPONENT)
+        # If we want legacy behaviour, directly call install with no meddling.
+        install(${ARGN})
+    else()
+        # We want to define the COMPONENT argument in the correct position, only if the user did not define
+        #  the COMPONENT argument. Therefore, capture the component argument and any arguments which can
+        #  legally follow it, so we can place those after the inserted COMPONENT argument.
+        set(options OPTIONAL EXCLUDE_FROM_ALL)
+        set(oneValueArgs COMPONENT RENAME)
+        # Specifying all valid first arguments as multiValueArgs captures all arguments between the first argument
+        #  and the COMPONENT argument (or any argument which can follow COMPONENT) in order.
+        set(multiValueArgs FILES PROGRAMS DIRECTORY CODE SCRIPT EXPORT FILES_MATCHING)
```

Comment:
Shouldn't `PROGRAMS` be in the runtime component not dev component?

Reply:
This is a good point. This also applies to `RUNTIME` (i.e. executable) items installed using `rocm_install_targets`, which should probably go in the runtime component rather than the dev component. Some libraries may generate executables that are only used for developing though, I will double check on that.

Reply:
This have both been implemented, although this may also need to be revisited in the future.